scripts/src/public/services/applications.ts

```diff
@@ -14,24 +14,28 @@ export type SignInContext = {
 	appUserId: string;
 };
 
-export type SpendPayloadOffer = {
+export type ExternalOfferPayload = {
 	id: string;
 	title: string;
 	description: string;
 	amount: number;
 	wallet_address: string;
 };
+export type EarnPayload = {
+	user_id: string;
+	offer: ExternalOfferPayload;
+};
 export type SpendPayload = {
-	offer: SpendPayloadOffer;
+	offer: ExternalOfferPayload;
 };
 
-export async function validateSpendJWT(jwt: string, logger: LoggerInstance): Promise<SpendPayloadOffer> {
-	const decoded = await verifyJWT<SpendPayload>(jwt, logger);
-	return decoded.payload.offer;
+export async function validateExternalOrderJWT(jwt: string, logger: LoggerInstance) {
+	const decoded = await verifyJWT<SpendPayload | EarnPayload, "spend" | "earn">(jwt, logger);
```

Comment:
This gives a change for an EarnPayload to arrive with "spend" subject. I prefer to be explicit here with the types.